cpp/tests/interop/arrow_utils.hpp

```diff
@@ -176,3 +176,27 @@ std::shared_ptr<arrow::Array> get_arrow_list_array(
 
 std::pair<std::unique_ptr<cudf::table>, std::shared_ptr<arrow::Table>> get_tables(
   cudf::size_type length = 10000);
+
+template <typename T>
+[[nodiscard]] auto make_decimal128_arrow_array(std::vector<T> const& data,
+                                               std::optional<std::vector<int>> const& validity,
```

Comment:
This is mostly a nit, but I'd prefer to see these be iterators, or at least spans.

Reply:
Because of the `.data()` I am not sure I can use iterators here. And `cudf::host_span` doesn't work with `std::vector`.